lib/charms/opensearch/v0/opensearch_peer_clusters.py

```diff
@@ -154,7 +151,21 @@ def run_with_relation_data(
             value=dict(sorted({node.name: node.to_dict() for node in data.cm_nodes}.items())),
             merge=True,
         )
-        self._charm.opensearch_config.add_seed_hosts([node.ip for node in data.cm_nodes])
+
+        # create alternative list for seed_hosts cluster-manager-only-nodes
+        hosts = []
+        if all_apps := self._charm.peers_data.get_object(Scope.APP, "cluster_fleet_apps"):
+            for app in all_apps.values():
+                p_cluster_app = PeerClusterApp.from_dict(app)
+                if "cluster_manager" not in p_cluster_app.roles:
+                    continue
+
+                hosts.append(p_cluster_app.leader_host)
```

Comment:
related to the previous question

Reply:
Has been removed, `cm_nodes` is now provided from the deployment desc and unit data if the nodes API is not available.